### PR TITLE
Add `race-check` target to Makefile. Closes #710

### LINKED ISSUE #710: Create a makefile target that checks number of races in e2e tests
# Problem

From time to time we found new data races in our code or in Ethereum. However, we haven't any process to clean up the code regularly.
We're migrating to Geth 1.8 and we don't know is it safe for us.

E.g we found a race in `les.downloader` and after it was fixed one more was raised https://github.com/status-im/status-go/issues/709#issuecomment-368921748

# Implementation

Make a `race-check` script that runs tests with `-race -v` flag and counts the number of discovered races. If that number is higher than expected number of races, it should fail. If less or equal — it should still succeed.

# Acceptance Criteria

1. `make race-check` is added to a Makefile
2. The script runs e2e tests with -race -v flag with a customizable `networkID`
3. There is a setting somewhere with the number of accepted (or known) races

# Future steps

Make a nightly run on Jenkins for the race check.

### PR DESCRIPTION
**Pull request by pedropombeiro**
Closes #710

### CODE REVIEW

**Review by JekaMas — APPROVED** [review]
Great! It's a huge step to have a stable application.

**Review by b00ris — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -21,11 +21,12 @@ XGOIMAGE = statusteam/xgo:$(XGOVERSION)
 XGOIMAGEIOSSIM = statusteam/xgo-ios-simulator:$(XGOVERSION)
 
 networkid ?= StatusChain
+e2e_extraflags =
```
Comment (dshulyak) [inline Makefile]: any reason not to use it for unit tests?

File: Makefile [hunk 2]
```diff
@@ -130,15 +131,18 @@ test-unit-coverage: ##@tests Run unit and integration tests with coverage
...
+	go test -timeout 5m ./t/e2e/node/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 50m ./t/e2e/jail/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 20m ./t/e2e/rpc/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 20m ./t/e2e/whisper/... -network=$(networkid) $(e2e_extraflags)
+	go test -timeout 10m ./t/e2e/transactions/... -network=$(networkid) $(e2e_extraflags)
 	# e2e_test tag is required to include some files from ./lib without _test suffix
-	go test -timeout 40m -tags e2e_test ./lib -network=$(networkid)
+	go test -timeout 40m -tags e2e_test ./lib -network=$(networkid) $(e2e_extraflags)
+
+race-check: e2e_extraflags=-race ##@tests Run e2e tests with -race flag
```
Comment (dshulyak) [inline Makefile]: a better place for my previous comment. why don't you want to use this flag for unit tests?

File: Makefile (hunk 1 above)
Comment (PombeirP) [reply Makefile]: Not at all. Are you suggesting we also add a `test-unit-race` target?

File: Makefile (hunk 2 above)
Comment (pedropombeiro) [reply Makefile]: Only because it requires a new target. If we want a new `test-unit-race` target, I can add it.

Comment (pedropombeiro) [reply Makefile]: @dshulyak or do you mean making the `race-check` target run both unit tests and e2e tests? That might actually make more sense. We won't have individual steps in the Jenkins pipeline though :-/

Comment (dshulyak) [reply Makefile]: both options looks good to me

**Review by dshulyak — APPROVED** [review]

**Pull request merged**